responses/registries.py

```diff
@@ -88,3 +88,26 @@ def find(
             return None, [reason]
 
         return response, []
+
+
+class LockRegistry(FirstMatchRegistry):
```

Comment:
Should this be something that the basic/default registry does? Having responses 'just work' by default would be nice and the overhead of locking a thread in environments without contention should be low.

Reply:
@markstory 
my only concern is the overhead since it looks like a rare case.

but if we are sure that this is negligible, then I can refactor